src/rgw/rgw_rest_callback_s3.h

```diff
@@ -0,0 +1,84 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef PROJECT_RGW_REST_CALLBACK_S3_H
+#define PROJECT_RGW_REST_CALLBACK_S3_H
+#include <errno.h>
+#include <string.h>
+
+#include "common/ceph_crypto.h"
+#include "common/safe_io.h"
+#include <boost/algorithm/string.hpp>
+#include <boost/algorithm/string/replace.hpp>
+
+#include <openssl/pem.h>
+
+#include "rgw_rest.h"
+#include "rgw_rest_s3.h"
+
+#include "rgw/rgw_b64.h"
+#include "include/assert.h"
+
+#define dout_subsys ceph_subsys_rgw
+
+class RGWCallBackClient : public RGWPostHTTPData {
```

Comment:
this is an alibaba extension--seems fine to support it, but other protocols/behaviors might be useful too